vispy/scene/widgets/colorbar.py

```diff
@@ -0,0 +1,144 @@
+        orientation can be thought of as the position of the label
+        relative to the color bar.
+
+        When the orientation is 'left' or 'right', the colorbar is
+        vertically placed. When it is 'top' or 'bottom', the colorbar is
+        horizontally placed. The colorbar automatically resizes when its
+        container's dimension changes.
+
+            * 'top': the colorbar is horizontal.
+              Color is applied from left to right.
+              Minimum corresponds to left and maximum to right.
+              Label is to the top of the colorbar
+
+            * 'bottom': Same as top, except that
+              label is to the bottom of the colorbar
+
+            * 'left': the colorbar is vertical.
+              Color is applied from bottom to top.
+              Minimum corresponds to bottom and maximum to top.
+              Label is to the left of the colorbar
+
+            * 'right': Same as left, except that the
+              label is placed to the right of the colorbar
+    label : str
+        The label that is to be drawn with the colorbar
+        that provides information about the colorbar.
+    clim : tuple (min, max)
+        the minimum and maximum values of the data that
+        is given to the colorbar. This is used to draw the scale
+        on the side of the colorbar.
+    border_width : float (in px)
+        The width of the border the colormap should have. This measurement
+        is given in pixels
+    border_color : str | vispy.color.Color
+        The color of the border of the colormap. This can either be a
+        str as the color's name or an actual instace of a vipy.color.Color
+    """
+    def __init__(self, cmap, orientation,
+                 label="", clim=("", ""),
+                 border_width=0.0, border_color="black", **kwargs):
+
+        dummy_size = (1, 1)
+        self._colorbar = ColorBarVisual(size=dummy_size, cmap=cmap,
+                                        orientation=orientation,
+                                        label=label, clim=clim,
+                                        border_width=border_width,
+                                        border_color=border_color, **kwargs)
+
+        Widget.__init__(self)
+
+        self.add_subvisual(self._colorbar)
+        self._update_colorbar()
+
+    def on_resize(self, event):
+        """Resize event handler
+
+        Parameters
+        ----------
+        event : instance of Event
+            The event.
+        """
+        self._update_colorbar()
+
+    def _update_colorbar(self):
+        self._colorbar.pos = self.rect.center
+        self._colorbar.size = \
+            ColorBarWidget.calc_size(self.rect, self._colorbar.orientation)
+
+    @staticmethod
+    def calc_size(rect, orientation):
+        (total_halfx, total_halfy) = rect.center
+
+        # the padding that the colorbar should leave with respect
+        # to the total halfdim
+        MAJOR_AXIS_PADDING = 0.1
```

Comment:
If it's local, use all lower-case. If it's a constant local, then it probably makes more sense to make it into an attribute so the user has some option to modify it (but I would not make these into properties because then we would require an API change if we decide to rethink the size management later on).